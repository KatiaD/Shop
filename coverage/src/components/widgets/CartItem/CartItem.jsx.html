<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/widgets/CartItem/CartItem.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/widgets/CartItem</a> CartItem.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'components/controls';
import { CartItemRow, CartItemCol, QuantityContainer } from './CartItem.styled';
&nbsp;
const displayName = 'CartItem';
&nbsp;
const propTypes = {
  id: PropTypes.number.isRequired,
};
&nbsp;
function <span class="fstat-no" title="function not covered" >CartItem(</span>{
  name,
  image,
  quantity,
  id,
  price,
  handlePlusItem,
  handleMinusItem,
  controls,
}) {
<span class="cstat-no" title="statement not covered" >  return (</span>
    id &amp;&amp; (
      &lt;CartItemRow data-name={id} key={id}&gt;
        &lt;CartItemCol&gt;&lt;img src={image} alt="" /&gt;&lt;/CartItemCol&gt;
        &lt;CartItemCol&gt;{name}&lt;/CartItemCol&gt;
        &lt;CartItemCol&gt;{price}&lt;/CartItemCol&gt;
        &lt;CartItemCol&gt;
          {
            controls ? (
              &lt;div&gt;
                &lt;Button onClick={handleMinusItem}&gt;-&lt;/Button&gt;
                &lt;QuantityContainer&gt;{quantity}&lt;/QuantityContainer&gt;
                &lt;Button onClick={handlePlusItem}&gt;+&lt;/Button&gt;
              &lt;/div&gt;
            ) : quantity
          }
        &lt;/CartItemCol&gt;
        &lt;CartItemCol&gt;
          {price * quantity}
        &lt;/CartItemCol&gt;
      &lt;/CartItemRow&gt;
    )
  );
}
&nbsp;
CartItem.displayName = displayName;
CartItem.propTypes = propTypes;
&nbsp;
export default CartItem;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Nov 27 2018 17:41:45 GMT+0200 (EET)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
