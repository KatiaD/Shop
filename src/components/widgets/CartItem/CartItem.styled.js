import styled from 'styled-components';

export const CartWrapper = styled.div`
  display: flex;
  justify-content: space-around;
`;

export const CartContainer = styled.div`
    display:flex;
  img {
    display: block;
    padding: 10px;

  }
`;
export const Price = styled.div`
  background: #fef2b8;
  border-radius: 4px;
  display: inline-block;
  padding: 7px 7px 5px;
  vertical-align: middle;
  margin-right: .5em;
  white-space: nowrap;
  border: 1px solid transparent;
  font-size: 14px;
`;

export const Name = styled.div`
  color: #212121;
  font-size: 14px;
  height: 30px;
  line-height: 15px;
  margin-bottom: 8px;
  overflow: hidden;
`;
