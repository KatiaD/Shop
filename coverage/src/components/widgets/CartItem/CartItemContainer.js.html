<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/widgets/CartItem/CartItemContainer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/widgets/CartItem</a> CartItemContainer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {
  compose, setDisplayName, withHandlers, setPropTypes,
} from 'recompose';
import PropTypes from 'prop-types';
import CartItem from './CartItem';
&nbsp;
&nbsp;
export const enhance = compose(
  setDisplayName('ProductContainer'),
  setPropTypes({
    id: PropTypes.number.isRequired,
    minus: PropTypes.func,
    plus: PropTypes.func,
    control: PropTypes.bool,
  }),
  withHandlers({
    handleMinusItem: ({ id, minus }) =&gt; () =&gt; minus(id),
    handlePlusItem: ({ id, plus }) =&gt; () =&gt; plus(id),
  }),
);
&nbsp;
export default enhance(CartItem);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Nov 27 2018 17:41:45 GMT+0200 (EET)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
