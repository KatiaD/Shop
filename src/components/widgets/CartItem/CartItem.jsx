import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'components/controls';

import { CartContainer, Price, Name } from './CartItem.styled';

const displayName = 'CartItem';

const propTypes = {
  id: PropTypes.string.isRequired,
};

function CartItem({
  name,
  image,
  quantity,
  id,
  price,
  handlePlusItem,
  handleMinusItem,
  controls
}) {
  return (
    id && (
      <CartContainer data-name={id} key={id}>
        <img src={image} />
        <Name>{name}</Name>
        <Price>{price}$</Price>
        <div>
          {
            controls ?
              <div>
                <Button onClick={handleMinusItem}>-</Button>
                {quantity}
                <Button onClick={handlePlusItem}>+</Button>
              </div>
              : quantity
          }
        </div>
        <Price>{price * quantity}$</Price>
      </CartContainer>
    )
  );
}

CartItem.displayName = displayName;
CartItem.propTypes = propTypes;

export default CartItem;
