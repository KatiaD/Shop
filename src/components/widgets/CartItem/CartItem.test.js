import React from 'react';
import { shallow } from 'enzyme';
import CartItem from './CartItem';

const testProps = {
  handlePlusItem: jest.fn(),
  handleMinusItem: jest.fn(),
  name: 'GUNMETAL SANDSTONE',
  id: 1,
  price: 140,
  image: '',
  quantity: 1,
  controls: true,
};

describe('Given the CartItem component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<CartItem {...testProps} />);
  });
  describe('when the component is rendered with controls', () => {
    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the component is rendered without controls', () => {
    beforeAll(() => {
      testProps.controls = false;
    });
    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
