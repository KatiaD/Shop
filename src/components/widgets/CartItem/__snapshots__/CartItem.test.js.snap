// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CartItem component when the component is rendered with controls should render the component with provided props 1`] = `
<CartItemstyled__CartItemRow
  data-name={1}
  key="1"
>
  <CartItemstyled__CartItemCol>
    <img
      alt=""
      src=""
    />
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    GUNMETAL SANDSTONE
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    140
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    <div>
      <Button
        name="minus"
        onClick={[MockFunction]}
      >
        -
      </Button>
      <CartItemstyled__QuantityContainer>
        1
      </CartItemstyled__QuantityContainer>
      <Button
        name="plus"
        onClick={[MockFunction]}
      >
        +
      </Button>
    </div>
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    140
  </CartItemstyled__CartItemCol>
</CartItemstyled__CartItemRow>
`;

exports[`Given the CartItem component when the component is rendered without controls should render the component with provided props 1`] = `
<CartItemstyled__CartItemRow
  data-name={1}
  key="1"
>
  <CartItemstyled__CartItemCol>
    <img
      alt=""
      src=""
    />
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    GUNMETAL SANDSTONE
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    140
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    1
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    140
  </CartItemstyled__CartItemCol>
</CartItemstyled__CartItemRow>
`;
